pyquil/device/_isa.py

```diff
@@ -162,19 +162,36 @@ def gates_in_isa(isa):
                 Gate("RX", [-np.pi], [unpack_qubit(q.id)]),
                 Gate("RZ", [THETA], [unpack_qubit(q.id)]),
             ])
+        elif q.type == "WILDCARD":
+            gates.extend([
+                Gate("_", "_", [unpack_qubit(q.id)])
+            ])
         else:  # pragma no coverage
             raise ValueError("Unknown qubit type: {}".format(q.type))
 
     for e in isa.edges:
         if e.dead:
             continue
         targets = [unpack_qubit(t) for t in e.targets]
-        if e.type in ["CZ", "ISWAP"]:
-            gates.append(Gate(e.type, [], targets))
-            gates.append(Gate(e.type, [], targets[::-1]))
-        elif e.type in ["CPHASE"]:
-            gates.append(Gate(e.type, [THETA], targets))
-            gates.append(Gate(e.type, [THETA], targets[::-1]))
+        edge_type = e.type if isinstance(e.type, list) else [e.type]
+        if "CZ" in edge_type:
+            gates.append(Gate("CZ", [], targets))
+            gates.append(Gate("CZ", [], targets[::-1]))
+        if "ISWAP" in edge_type:
+            gates.append(Gate("ISWAP", [], targets))
+            gates.append(Gate("ISWAP", [], targets[::-1]))
+        if "XYhalves" in edge_type:
+            gates.append(Gate("XY", [np.pi/2], targets))
+            gates.append(Gate("XY", [np.pi/2], targets[::-1]))
+        if "CPHASE" in edge_type:
+            gates.append(Gate("CPHASE", [THETA], targets))
+            gates.append(Gate("CPHASE", [THETA], targets[::-1]))
+        if "XY" in edge_type:
+            gates.append(Gate("XY", [THETA], targets))
+            gates.append(Gate("XY", [THETA], targets[::-1]))
+        if e.type in ["WILDCARD"]:
```

Comment:
why is this one different instead of `if "WILDCARD" in edge_type` ?

Reply:
bad copy/paste